### PR TITLE
Upgrading python to 3.8 and remove automl pinned packages

### PR DESCRIPTION
**Pull request by imatiach-msft**

### CODE REVIEW

File: responsibleai/environments/responsibleai-vision-ubuntu20.04-py38-cpu/context/Dockerfile [hunk 1]
```diff
@@ -20,10 +20,10 @@ RUN conda env create -p $AZUREML_CONDA_ENVIRONMENT_PATH -f conda_dependencies.ya
 RUN pip install --pre 'azure-ai-ml' 'azure-storage-blob<=12.13.0' 'numpy<1.24.0'
 
 # Install azureml packages
-RUN pip install 'azureml-dataset-runtime==1.48.0' \
-                'azureml-core==1.48.0' \
-                'azureml-mlflow==1.48.0' \
-                'azureml-automl-dnn-vision==1.48.0' \
+RUN pip install 'azureml-dataset-runtime' \
```
Comment (vizhur) [inline responsibleai/environments/responsibleai-vision-ubuntu20.04-py38-cpu/context/Dockerfile:23]: do you want to resolve it to the latest available version instead of dropping the pin?

**Review by vinuthakaranth — APPROVED** [review]

**Review by tongyu-microsoft — DISMISSED** [review]

File: responsibleai/environments/responsibleai-vision-ubuntu20.04-py38-cpu/context/Dockerfile (hunk 1 above)
Comment (imatiach-msft) [reply responsibleai/environments/responsibleai-vision-ubuntu20.04-py38-cpu/context/Dockerfile:23]: I will set it explicitly to 1.49.0 because I don't think this logic works for Dockerfile, just conda_dependencies.yaml files:

https://github.com/Azure/azureml-assets/blob/main/scripts/azureml-assets/azureml/assets/environment/pin_package_versions.py#L19

**Review by romanlutz — DISMISSED** [review]

**Review by xuke444 — APPROVED** [review]

File: responsibleai/environments/responsibleai-vision-ubuntu20.04-py38-cpu/context/Dockerfile (hunk 1 above)
Comment (imatiach-msft) [reply responsibleai/environments/responsibleai-vision-ubuntu20.04-py38-cpu/context/Dockerfile]: update: it seems it can't find the latest package version of automl-dnn-vision package, there is some error:

2023-02-27T21:42:12.4536065Z ##[debug]Looking up latest version of azureml-dataset-runtime
2023-02-27T21:42:12.5459201Z ##[debug]Latest version of azureml-dataset-runtime is 1.49.0
2023-02-27T21:42:12.5461662Z ##[debug]Looking up latest version of azureml-core
2023-02-27T21:42:12.5933374Z ##[debug]Latest version of azureml-core is 1.49.0
2023-02-27T21:42:12.5936807Z ##[debug]Looking up latest version of azureml-mlflow
2023-02-27T21:42:12.6121314Z ##[debug]Latest version of azureml-mlflow is 1.49.0
2023-02-27T21:42:12.6124209Z ##[debug]Looking up latest version of azureml-automl-dnn-vision
2023-02-27T21:42:12.6240223Z ##[debug]Latest version of azureml-automl-dnn-vision is None

I am trying to debug the reason why it is set to None for some reason

Comment (elesel) [reply responsibleai/environments/responsibleai-vision-ubuntu20.04-py38-cpu/context/Dockerfile]: It's not finding azureml-automl-dnn-vision because the GitHub agent on which the lookup is performed is running Python 3.11 and [azureml-automl-dnn-vision](https://pypi.org/project/azureml-automl-dnn-vision/) requires Python < 3.9.

Maybe it's best to limit the agent to Python 3.8 for now or find a good way to make the search Python version agnostic.

Comment (elesel) [reply responsibleai/environments/responsibleai-vision-ubuntu20.04-py38-cpu/context/Dockerfile]: Here's an approach that I believe will work: #413

Comment (elesel) [reply responsibleai/environments/responsibleai-vision-ubuntu20.04-py38-cpu/context/Dockerfile]: @imatiach-msft, would you please merge the main branch into yours? #413 was just merged.

File: scripts/azureml-assets/azureml/assets/environment/pin_package_versions.py [hunk 2]
```diff
@@ -23,6 +23,8 @@
 
 def create_package_finder(index_urls: List[str]) -> PackageFinder:
     """Create a pip PackageFinder."""
+    print("List of index urls: " + str(index_urls))
+    logger.log_debug("List of index urls: " + str(index_urls))
```
Comment (elesel) [inline scripts/azureml-assets/azureml/assets/environment/pin_package_versions.py:27]: Please revert the changes to this file after you're done testing/debugging.

File: responsibleai/environments/responsibleai-vision-ubuntu20.04-py38-cpu/context/Dockerfile (hunk 1 above)
Comment (imatiach-msft) [reply responsibleai/environments/responsibleai-vision-ubuntu20.04-py38-cpu/context/Dockerfile]: done! The package installation steps seem to be working now.

File: scripts/azureml-assets/azureml/assets/environment/pin_package_versions.py (hunk 2 above)
Comment (imatiach-msft) [reply scripts/azureml-assets/azureml/assets/environment/pin_package_versions.py]: done! reverted the changes to the file.

**Review by elesel — APPROVED** [review]

**Review by romanlutz — DISMISSED** [review]

**Review by tongyu-microsoft — APPROVED** [review]

**Review by gaugup — APPROVED** [review]

File: responsibleai/environments/responsibleai-vision-ubuntu20.04-py38-cpu/context/Dockerfile [hunk 3]
```diff
@@ -20,10 +20,10 @@ RUN conda env create -p $AZUREML_CONDA_ENVIRONMENT_PATH -f conda_dependencies.ya
...
 
 # Install azureml packages
-RUN pip install 'azureml-dataset-runtime==1.48.0' \
-                'azureml-core==1.48.0' \
-                'azureml-mlflow==1.48.0' \
-                'azureml-automl-dnn-vision==1.48.0' \
+RUN pip install 'azureml-dataset-runtime=={{latest-pypi-version}}' \
+                'azureml-core=={{latest-pypi-version}}' \
+                'azureml-mlflow=={{latest-pypi-version}}' \
+                'azureml-automl-dnn-vision=={{latest-pypi-version}}' \
```
Comment (gaugup) [inline responsibleai/environments/responsibleai-vision-ubuntu20.04-py38-cpu/context/Dockerfile:26]: You can remove the extra `\` in line 26.
